client/my-sites/comments/comment-list/index.jsx

```diff
@@ -88,21 +89,28 @@ export class CommentList extends Component {
 			this.setState( {
 				isBulkEdit: false,
 				lastUndo: null,
-				page: 1,
 				persistedComments: [],
 				selectedComments: [],
 			} );
 		}
 	}
 
-	changePage = page => {
-		const total = Math.ceil( ( this.props.comments.length + this.state.persistedComments.length ) / COMMENTS_PER_PAGE );
-		this.props.recordChangePage( page, total );
+	changePage = pageNumber => {
+		const {
+			recordChangePage,
+			siteFragment,
+			status,
+		} = this.props;
 
-		this.setState( {
-			page,
-			selectedComments: [],
-		} );
+		recordChangePage( page, this.getTotalPages() );
+
+		this.setState( { selectedComments: [] } );
+
+		if ( window ) {
+			window.scrollTo( 0, 0 );
+		}
+
+		page( addQueryArgs( { page: pageNumber }, `/comments/${ status }/${ siteFragment }` ) );
```

Comment:
please remember to think of the children! this is a bunch of window-specific side-effecting code. how can we test it? how can we verify it?

could we imagine a way to pass this stuff in as `props` in a way that lets us change the page without the side-effects, but preserve them in Calypso when it counts?